File: backend/Models/Destination.cs
namespace Travyle.Api.Models;

public class Destination
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Region { get; set; } = string.Empty; // Country / province
    public string Description { get; set; } = string.Empty;
    
    public string[] Tags { get; set; } = Array.Empty<string>(); // Searchable tags
    
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    
    public string ImageUrl { get; set; } = string.Empty; // Supabase Storage URL
    
    public decimal AverageRating { get; set; } = 0.0m; // Computed from reviews (Student 4)
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: mobile/lib/features/destinations/models/destination.dart
class Destination {
  final String id;
  final String name;
  final String region;
  final String description;
  final List<String> tags;
  final String imageUrl;
  final double averageRating;

  const Destination({
    required this.id,
    required this.name,
    required this.region,
    required this.description,
    required this.tags,
    required this.imageUrl,
    required this.averageRating,
  });

  factory Destination.fromJson(Map<String, dynamic> json) {
    return Destination(
      id: json['id'].toString(),
      name: json['name'] as String? ?? '',
      region: json['region'] as String? ?? '',
      description: json['description'] as String? ?? '',
      tags: List<String>.from(json['tags'] as List? ?? const []),
      imageUrl: json['imageUrl'] as String? ?? '',
      averageRating: (json['averageRating'] as num?)?.toDouble() ?? 0,
    );
  }
}
